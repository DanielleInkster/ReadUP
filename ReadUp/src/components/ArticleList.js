import React from 'react';
import ArticleView from './ArticleView';
import {FlatList} from 'react-native';

const ArticleList = ({deleteEntry, articles}) => {
  return (
    <FlatList
      data={articles}
      renderItem={({item}) => (
        <ArticleView article={item} deleteEntry={deleteEntry} />
      )}
    />
  );
};
export default ArticleList;
