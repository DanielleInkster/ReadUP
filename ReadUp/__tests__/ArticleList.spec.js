import React from 'react';
import {render} from 'react-native-testing-library';
import ArticleList from '../src/components/ArticleList';

import renderer from 'react-test-renderer';

const articles = [
  {id: '1', title: 'hello'},
  {id: '2', title: 'world'},
];
const fakeCall = jest.fn();

test('renders correctly', () => {
  const tree = renderer
    .create(<ArticleList articles={articles} deleteEntry={fakeCall} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

describe('ArticleList', () => {
  describe('list', () => {
    it('generates a list of articles', () => {
      const componentTree = render(
        <ArticleList articles={articles} deleteEntry={fakeCall} />,
      );
      expect(componentTree.getAllByTestId('listItem').length).toBe(
        articles.length,
      );
    });
  });
});
