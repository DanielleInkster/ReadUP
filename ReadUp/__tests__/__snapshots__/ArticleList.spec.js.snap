// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "id": "1",
        "title": "hello",
      },
      Object {
        "id": "2",
        "title": "world",
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#bbe1fa",
            "borderBottomWidth": 1,
            "borderColor": "#eee",
            "borderRadius": 10,
            "marginHorizontal": "2%",
            "marginVertical": "3%",
            "opacity": 1,
            "padding": 20,
          }
        }
        testID="listItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#1b262c",
                "fontFamily": "sans-serif-medium",
                "fontSize": 18,
              }
            }
          >
            hello
            ...
          </Text>
          <Text
            allowFontScaling={false}
            onPress={[Function]}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Object {
                  "color": "firebrick",
                  "fontSize": 25,
                },
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
            testID="removeArticle"
          >
            
          </Text>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#bbe1fa",
            "borderBottomWidth": 1,
            "borderColor": "#eee",
            "borderRadius": 10,
            "marginHorizontal": "2%",
            "marginVertical": "3%",
            "opacity": 1,
            "padding": 20,
          }
        }
        testID="listItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#1b262c",
                "fontFamily": "sans-serif-medium",
                "fontSize": 18,
              }
            }
          >
            world
            ...
          </Text>
          <Text
            allowFontScaling={false}
            onPress={[Function]}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Object {
                  "color": "firebrick",
                  "fontSize": 25,
                },
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
            testID="removeArticle"
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
